docs/content/preview/contribute/core-database/merge-with-upstream-repositories.md

```diff
@@ -0,0 +1,1017 @@
+In particular, references to "Phorge", "`arc`", and "internal slack" are internal.
+Such references generally pertain to putting up [yugabyte/yugabyte-db][repo-yugabyte-db] code for review, getting it tested through jenkins, and committing it.
+External contributors may publish the code for review using a GitHub PR, and an internal engineer will take care of the internal steps.
+External contributors may also reach out in our [community Slack]({{<slack-invite>}}).
+
+## Different tracking approaches
+
+In the beginning, there was no formality importing embedding external code.
+Examples include RocksDB and Kudu.
+Such code is now so closely entwined with YugabyteDB's modifications such that it is difficult to update them to newer versions.
+
+PostgreSQL also technically fell under that bucket.
+However, today, it is in a better state with formality imposed onto it.
+That formality primarily comes from [`upstream_repositories.csv`][upstream-repositories-csv].
+
+### How upstream repositories are tracked
+
+[`upstream_repositories.csv`][upstream-repositories-csv] specifies which commit is being tracked in each upstream repository.
+(Naturally, this means that it will not work for external code that is not in the form of a git repository, but in this day and age, that is not likely to happen.)
+This makes it apparent what changes YugabyteDB made against the upstream repository.
+
+In case the commit being tracked is present in the external repository, this may seem trivial to do anyway without [`upstream_repositories.csv`][upstream-repositories-csv].
+However, YugabyteDB occasionally takes point-imports of external repository commits, such as security fixes.
+To make matters worse, those point-imports can have conflicts on the external repository side, and they could have other conflicts on the [yugabyte-db repository][repo-yugabyte-db] side as well.
+Do that ten or so times, and it becomes difficult to track which changes are made by YugabyteDB and which changes are point-imported.
+The CSV necessitates a single commit tracking what upstream state is being tracked.
+Therefore, in case of port-imports, YugabyteDB must create a fork of the external repository to track the state of the upstream code with the point-imports.
+
+To make matters more complicated, the upstream repository might be tracked differently across [yugabyte/yugabyte-db][repo-yugabyte-db] branches.
+For example, pgaudit has different branches for each PG version while pg_cron has a single branch where, at any point, there is multi-version support.
+In case of the former, it is necessary to track separate upstream commits across [yugabyte/yugabyte-db][repo-yugabyte-db] branches, and if a fork is required, then a branch should be created for each PG version that YugabyteDB supports, conventionally named `yb-pg<version>`.
+In case of the latter, one may do the same, or a single `yb` branch can be used across all [yugabyte/yugabyte-db][repo-yugabyte-db] branches.
```

Comment:
```suggestion
In case of the latter, you may do the same, or a single `yb` branch can be used across all [yugabyte/yugabyte-db][repo-yugabyte-db] branches.
```